recipe/meta.yaml

```diff
@@ -651,8 +665,168 @@ outputs:
         - numpy                                  # [build_platform != target_platform]
         - cmake
         - ninja
+      host:
+        - {{ pin_subpackage("libarrow", exact=True) }}
```

Comment:
@h-vetinari @xhochy that's my understanding on why they have to be on host, let me know your thoughts:
```suggestion
      host:
        # We add all libarrow package dependencies on host in order to build pyarrow
        # once with all capabilities. We require the exact hash versions but we don't
        # want those to be installed when running for pyarrow-core as we want a
        # low memory footprint for the pyarrow-core package.
        # We cannot use them on build as we want the exact hash version built on the 
        # recipe to be linked, otherwise we get mismatching package errors.
        - {{ pin_subpackage("libarrow", exact=True) }}
```
as an alternative we can add them as build dependencies with:
```diff
diff --git a/recipe/meta.yaml b/recipe/meta.yaml
index 749176b..29ba9fb 100644
--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -665,15 +665,15 @@ outputs:
         - numpy                                  # [build_platform != target_platform]
         - cmake
         - ninja
+        - {{ pin_subpackage("libarrow-acero") }}
+        - {{ pin_subpackage("libarrow-dataset") }}
+        - {{ pin_subpackage("libarrow-flight") }}
+        - {{ pin_subpackage("libarrow-flight-sql") }}
+        - {{ pin_subpackage("libarrow-gandiva") }}
+        - {{ pin_subpackage("libarrow-substrait") }}
+        - {{ pin_subpackage("libparquet") }}
       host:
         - {{ pin_subpackage("libarrow", exact=True) }}
-        - {{ pin_subpackage("libarrow-acero", exact=True) }}
-        - {{ pin_subpackage("libarrow-dataset", exact=True) }}
-        - {{ pin_subpackage("libarrow-flight", exact=True) }}
-        - {{ pin_subpackage("libarrow-flight-sql", exact=True) }}
-        - {{ pin_subpackage("libarrow-gandiva", exact=True) }}
-        - {{ pin_subpackage("libarrow-substrait", exact=True) }}
-        - {{ pin_subpackage("libparquet", exact=True) }}
         - clangdev {{ llvm_version }}
         - llvmdev {{ llvm_version }}
         - cython
```
but from my understanding we require `exact=True` to match the one build on the recipe but again, my conda knowledge is limited and I might be doing something wrong here.

Reply:
No, they shouldn't be in build. But the main part you already answered. It's about keeping the footprint of `pyarrow` small - which now also has graceful fallbacks in case the libs aren't found - and leaving the full-fledged install to `pyarrow-all`.